### PR TITLE
From fetch to axios interceptor

### PR DESCRIPTION
**Pull request by steven-dev1**

### CODE REVIEW

File: infrastructure/utils/fetchers.ts [hunk 1]
```diff
@@ -1,81 +1,63 @@
...
 
-      const data = await response?.json()
-      return data
-  } catch (error) {
-      return console.error('Error al obtener los favoritos:', error);
+export const fetchFavorites = async (sessionId: string) => {
+  if (!sessionId) return 'No session id provided';
+  
+  const response = await apiClient.post('/favorites', {
+    sessionId: sessionId,
```
Comment (Surialpha) [inline infrastructure/utils/fetchers.ts:18]: If you intercept request, you should not send it back

File: infrastructure/utils/fetchers.ts [hunk 2]
```diff
@@ -1,81 +1,63 @@
...
-  try {
-    const response = await fetch(`/api/favorites`, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({ 'match_id': id, sessionId, 'action': httpPostActions.POST_FAVORITES }),
+    const response = await apiClient.post('/favorites', {
+      match_id: id,
+      sessionId: sessionId,
```
Comment (Surialpha) [inline infrastructure/utils/fetchers.ts:37]: same here

File: infrastructure/utils/fetchers.ts [hunk 3]
```diff
@@ -1,81 +1,63 @@
...
   try {
-    const response = await fetch(`/api/favorites`, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({ 'match_id': id, sessionId, 'action': httpPostActions.DELETE_FAVORITES }),
+    const response = await apiClient.post(`/favorites`, {
+      match_id: id,
+      sessionId: sessionId,
```
Comment (Surialpha) [inline infrastructure/utils/fetchers.ts:53]: and here

File: interceptors/axios.interceptors.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+
+const apiClient = axios.create(
+    {
+        baseURL: '/api'
+    }
+)
+apiClient.interceptors.request.use(
+    (config) => {
+        const sessionId = config.data?.sessionId
```
Comment (Surialpha) [inline interceptors/axios.interceptors.tsx:10]: Use the local storage here if posible

Comment (steven-dev1) [reply interceptors/axios.interceptors.tsx:10]: I get an error cause localStorage is not defined

**Review by Surialpha — APPROVED** [review]

**Pull request merged**